test/integration/cronjob/cronjob_test.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronjob
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	cacheddiscovery "k8s.io/client-go/discovery/cached"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	clientbatchv1beta1 "k8s.io/client-go/kubernetes/typed/batch/v1beta1"
+	"k8s.io/client-go/rest"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/kubernetes/pkg/controller/cronjob"
+	"k8s.io/kubernetes/pkg/controller/garbagecollector"
+	"k8s.io/kubernetes/pkg/controller/job"
+	"k8s.io/kubernetes/pkg/util/metrics"
+	"k8s.io/kubernetes/test/integration/framework"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func setup(t *testing.T) (*httptest.Server, framework.CloseFunc, *cronjob.CronJobController, *job.JobController, informers.SharedInformerFactory, clientset.Interface, rest.Config) {
+	masterConfig := framework.NewIntegrationTestMasterConfig()
+	_, server, closeFn := framework.RunAMaster(masterConfig)
+
+	config := restclient.Config{Host: server.URL}
+	clientSet, err := clientset.NewForConfig(&config)
+	if err != nil {
+		t.Fatalf("Error creating clientset: %v", err)
+	}
+	resyncPeriod := 12 * time.Hour
+	informerset := informers.NewSharedInformerFactory(clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "cronjob-informers")), resyncPeriod)
+	metrics.UnregisterMetricAndUntrackRateLimiterUsage("cronjob_controller")
+	cjc, err := cronjob.NewCronJobController(clientSet)
+	if err != nil {
+		t.Fatalf("Error creating CronJob controller: %v", err)
+	}
+	jc := job.NewJobController(informerset.Core().V1().Pods(), informerset.Batch().V1().Jobs(), clientSet)
+
+	return server, closeFn, cjc, jc, informerset, clientSet, config
+}
+
+func setupGarbageCollector(t *testing.T, config rest.Config, clientSet clientset.Interface, informerset informers.SharedInformerFactory, stopCh chan struct{}) {
```

Comment:
Why we need to setup GC for this test? From my pov this only complicates this test code.